src/pages/instances/actions/AttachIsoBtn.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { FC, useState } from "react";
+import { updateInstance } from "api/instances";
+import { LxdInstance } from "types/instance";
+import { useParams } from "react-router-dom";
+import { useNotify } from "@canonical/react-components";
+import { getInstanceEditValues, getInstancePayload } from "util/instanceEdit";
+import { LxdIsoDevice } from "types/device";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import usePortal from "react-useportal";
+import { RemoteImage } from "types/image";
+import SubmitButton from "components/SubmitButton";
+import CustomIsoModal from "pages/images/CustomIsoModal";
+
+interface Props {
+  instance: LxdInstance;
+}
+
+const AttachIsoBtn: FC<Props> = ({ instance }) => {
+  const { project } = useParams<{ project: string }>();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const [isLoading, setLoading] = useState(false);
+
+  const attachedIso = instance.devices["iso-volume"] as
+    | LxdIsoDevice
+    | undefined;
+
+  const ejectIso = () => {
+    setLoading(true);
+    const values = getInstanceEditValues(instance);
+    values.devices = values.devices.filter((device) => {
+      return device.name !== "iso-volume";
+    });
+    const instanceMinusIso = getInstancePayload(
+      instance,
+      values
+    ) as LxdInstance;
+    updateInstance(instanceMinusIso, project ?? "")
+      .then(() =>
+        notify.success(
+          <>
+            ISO <b>{attachedIso?.source ?? ""}</b> ejected
+          </>
+        )
+      )
+      .catch((e: Error) => notify.failure("Detach ISO failed", e))
+      .finally(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances, instance.name, project],
+        });
+        setLoading(false);
+      });
+  };
+
+  const handleSelect = (image: RemoteImage) => {
+    setLoading(true);
+    closePortal();
+    const values = getInstanceEditValues(instance);
+    values.devices.push({
+      type: "iso-volume",
+      name: "iso-volume",
+      bare: {
+        "boot.priority": "10",
+        pool: image.pool ?? "",
+        source: image.aliases,
+        type: "disk",
+      },
+    });
```

Comment:
This chunk of code is currently used in 2 different places - here and in `CreateInstanceForm`. Maybe it would be good to create a reusable helper function for it?

Reply:
Very good point! I wanted to extract this but forgot -- will do that now!